postgres/src/rows.rs

```diff
@@ -259,13 +256,21 @@ impl<'a> Row<'a> {
 pub trait RowIndex {
     /// Returns the index of the appropriate column, or `None` if no such
     /// column exists.
-    fn idx(&self, stmt: &Statement) -> Option<usize>;
+    fn idx<I: AsRef<StatementInfo>>(&self, info: &I) -> Option<usize> {
```

Comment:
Making this change to `RowIndex` is still backwards incompatible to implementors. My gut says that just changing the trait will cause any issues in practice though since it's basically an implementation detail.